### PR TITLE
Add a prop to expose a DOM ref on input to parent component

### PR DESCRIPTION
**Pull request by hannaliebl**
### Added
Added `fieldRef` prop to `TextField` to expose a ref on the input/textarea.

### CODE REVIEW

**Comment by sawyerh** [comment]
@hannaliebl Looks good 👍  

What were the issues you mentioned having with unit tests? Anything I can help with?

**Comment by hannaliebl** [comment]
Just not sure if a unit test should be added for this, and if so, how it would look. I guess there's not that much to actually test because the implementation can vary so much.

**Comment by sawyerh** [comment]
@hannaliebl Ah got it. I think having a unit test to confirm this returns a reference to the expected component would be nice. One way we could do that is by setting the `defaultValue` prop and then checking the fieldRef's `value` to confirm it matches. That would verify that `fieldRef` returned a reference to the `input`.

File: packages/core/src/components/TextField/TextField.test.jsx [hunk 1]
```diff
@@ -90,44 +91,59 @@ describe('TextField', function() {
...
     expect(data.wrapper.find('FormLabel').hasClass('bar')).toBe(true);
   });
 
+  it('returns reference to input field', () => {
+    let ref;
+    const data = render(
+      {
+        defaultValue: 'Yay',
+        fieldRef: el => {
+          ref = el;
```
Comment (sawyerh) [inline packages/core/src/components/TextField/TextField.test.jsx]: @hannaliebl Here's how I got this to work. This is similar to how this prop would be used within a real world component, ie:

```jsx
<TextField fieldRef={el => { this.input = el } } />
```

**Pull request merged**